src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,37 @@
 package mate.academy;
 
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final List<EventListener> listeners = new CopyOnWriteArrayList<>();
+    private final ExecutorService executorService = Executors.newFixedThreadPool(5);
 
+    public void registerListener(EventListener listener) {
+        listeners.add(listener);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        listeners.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        for (EventListener listener : listeners) {
+            executorService.submit(() -> listener.onEvent(event));
+        }
     }
```

Comment:
Submitting tasks to the executorService inside a loop can potentially create a large number of tasks in a short amount of time. If the number of events to notify is large, this could lead to performance issues or resource exhaustion. Consider using a more controlled way of executing tasks, possibly with rate limiting or using a bounded queue with a ThreadPoolExecutor to handle excessive load.